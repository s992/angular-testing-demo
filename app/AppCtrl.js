(function() {
	"use strict";

	angular.module("app.controllers")

	.controller("AppCtrl", ["$scope", "$location", "AppService", function( $scope, $location, AppService ) {

		AppService.getUser( $location.search().userId ).then(function( user ) {
			$scope.user = user;
		});

		$scope.clearUser = function() {
			$scope.user = undefined;
		};

	}]);
})();