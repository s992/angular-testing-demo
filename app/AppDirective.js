(function() {
	"use strict";

	angular.module("app.directives")

	.directive("header", function() {
		return {
			restrict: "E",
			scope: {
				text: "@"
			},
			template: "<h1>{{ text }}</h1>"
		};
	}]);
});