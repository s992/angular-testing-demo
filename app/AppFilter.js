(function() {
	"use strict";

	angular.module("app.filters")

	.filter("truncate", function() {
		return function( value, maxLength ) {
			if( value.length <= maxLength ) {
				return value;
			}

			return value.substr( 0, maxLength ) + "...";
		};
	});
})();