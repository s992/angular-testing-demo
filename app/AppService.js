(function() {
	"use strict";

	angular.module("app.services")

	.factory("AppService", ["$http", "apiUrl", function( $http, apiUrl ) {
		var service = {};

		service.getUser = function( userId ) {
			return $http.get( apiUrl + "/user/" + userId );
		};

		return service;
	}]);
});