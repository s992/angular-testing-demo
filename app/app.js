(function() {
	"use strict";

	var app = angular.module("app", [
		"app.controllers",
		"app.services",
		"app.directives",
		"app.filters"
	]);

	app.constant("apiUrl", "http://localhost/api");

	angular.module("app.controllers", []);
	angular.module("app.services", []);
	angular.module("app.directives", []);
	angular.module("app.filters", []);
})();