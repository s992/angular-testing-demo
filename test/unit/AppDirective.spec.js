require("../../app/app");
require("../../app/AppDirective");

describe("Header", function() {

	var element,
		scope,
		$compile;

	beforeEach(function() {
		angular.mock.module("app.directives", function( $provide ) {});

		inject(function( $injector ) {
			scope = $injector.get("$rootScope").$new();
			$compile = $injector.get("$compile");

			element = angular.element("<header text='headerText'></header>");
			scope.headerText = "Hello!";
			$compile( element )( scope );
			scope.$apply();
		});
	});

	it("should match the snapshot", function() {
		expect( element.html() ).toMatchSnapshot();
	});

	it("should create an h1 element", function() {
		expect( element.children("h1") ).toBeDefined();
	});

	it("should insert the text into the h1 tag", function() {
		expect( element.children("h1").text() ).toBe("Hello!");
	});

});