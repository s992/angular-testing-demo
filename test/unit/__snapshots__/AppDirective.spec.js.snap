exports[`Header should match the snapshot 1`] = `"<h1 class=\"ng-binding\">Hello!</h1>"`;
